pyface/ui/qt4/tasks/advanced_editor_area_pane.py

```diff
@@ -315,7 +316,7 @@ def compare(one, two):
                  (isinstance(child, QtGui.QDockWidget) and
                   (visible_only or not self.tabifiedDockWidgets(child))))):
                 children.append(child)
-        children.sort(cmp=compare)
+        children.sort(key=functools.cmp_to_key(compare))
```

Comment:
This is converting `key` to `cmp` function back to `key`. From the `compare` function definition above it looks like the `key` can simply be: `lambda widget: tuple(widget.pos().y(), widget.pos().x())`